src/FilterContext.ts

```diff
@@ -0,0 +1,22 @@
+// (C) 2019 GoodData Corporation
+
+export type FilterContextItem = IAttributeFilter | IDateFilter;
+
+export interface IAttributeFilter {
+    attributeFilter: {
+        displayForm: string;
+        negativeSelection: boolean;
+        attributeElements: string[];
+    };
+}
+
+export interface IDateFilter {
+    dateFilter: {
+        type: 'relative' | 'absolute';
+        granularity: string;
+        dataSet?: string;
+        from?: string;
```

Comment:
In both relative and absolute type, `from`, `to` is needed. Which case can `from` and `to` be missing?

Reply:
For `relative`, this year or today means `from=to=0`, so we can omit it.
For `absolute`, it means from/to point to `current` time.
Anyway, FE should send this info explicitly to BE.
Fixed.